src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,399 @@
+    private func findIntensity(lat: Double, long: Double, n: Double) -> [Double] {
+        var numerator: Double = 0
+        var denominator: Double = 0
+        for point in self.data {
+            let dist = self.distance(
+                lat1: lat,
+                long1: long,
+                lat2: point[0],
+                long2: point[1]
+            )
+            let distanceWeight = pow(dist, Double(n))
+            if distanceWeight == 0 {
+                continue
+            }
+            numerator += (point[2] / distanceWeight)
+            denominator += (1 / distanceWeight)
+        }
+        return [numerator, denominator]
+    }
+    
+    /// A helper function that finds the minimum and maximum longitude and latitude values that still contains a powerful enough
+    /// intensity that it should be included in the data set
+    ///
+    /// - Parameter input: A list of points that are in a cluster.
+    /// - Returns: A list of four integers representing the minimum and maximum longitude and latitude values
+    private func findBounds(input: [CLLocationCoordinate2D], n: Double) -> [Int] {
+        
+        // Initialize the boundary values to something that must be updated immediately
+        // 0: min lat, 1: min long, 2: max lat, 3: max long
+        var ans = [0x7fffffff, 0x7fffffff, -0x7fffffff, -0x7fffffff]
+        for coord in input {
+            ans[0] = min(ans[0], Int(coord.latitude * 10))
+            ans[1] = min(ans[1], Int(coord.longitude * 10))
+            ans[2] = max(ans[2], Int(coord.latitude * 10))
+            ans[3] = max(ans[3], Int(coord.longitude * 10))
+        }
+        
+        // A copy of the answer array is needed since the answer array is directly altered in the
+        // following while loops; the original values need to be retained to calculate
+        let copy = ans
+        
+        // The following while statements find the maximum and minimum lat/long values where the
+        // points are intense enough to be placed in the heat map's data set
+        
+        // Finds the minimum latitude
+        while (ans[0] > -900) {
+            let intensity = findIntensity(
+                lat: Double(ans[0]) / 10,
+                long: Double((ans[3] + ans[1]) / 2) / 10,
+                n: n
+            )
+            if intensity[1] == 0 || intensity[0] < 4 {
+                break
+            }
+            ans[0] -= 1
+        }
+        
+        // Finds the minimum longitude
+        while (ans[1] > -900) {
+            let intensity = findIntensity(
+                lat: Double((copy[2] + copy[0]) / 2) / 10,
+                long: Double(ans[1]) / 10,
+                n: n
+            )
+            if intensity[1] == 0 || intensity[0] < 4 {
+                break
+            }
+            ans[1] -= 1
+        }
+        
+        // Finds the maximum latitude
+        while (ans[2] < 1800) {
```

Comment:
How was 1800 determined?

Reply:
Also fixed this; there was a small issue.